hail/python/hail/expr/expressions/typed_expressions.py

```diff
@@ -3840,15 +3845,35 @@ def __getitem__(self, item):
                         hl.str("Index ") + hl.str(s) + hl.str(f" is out of bounds for axis {i} with size ") + hl.str(dlen)
                     )
                     slices.append(checked_int)
-            return construct_expr(ir.NDArraySlice(self._ir, hl.tuple(slices)._ir),
-                                  tndarray(self._type.element_type, n_sliced_dims),
-                                  self._indices,
-                                  self._aggregations)
+            product_item = construct_expr(ir.NDArraySlice(self._ir, hl.tuple(slices)._ir),
```

Comment:
Let's call `product_item` `sliced_expr`, less easy to confuse with `item`, which is the input to `__getitem__`.